pkg/apis/db/v1alpha1/db.go

```diff
@@ -0,0 +1,47 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Database is a specification for a Database resource
+type Database struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   DatabaseSpec   `json:"spec"`
+	Status DatabaseStatus `json:"status"`
+}
+
+// DatabaseSpec is the spec for a DatabaseServer resource
+type DatabaseSpec struct {
+	UserList []DatabaseUser          `json:"userList"`
+	DsnFrom  *corev1.SecretReference `json:"dsnFrom"`
```

Comment:
I cannot see any reason to have specific plugin name here (it is in README). If it will be needed in some case, there should be ref to parent resource which contains the plugin instance.